components/google_drive/actions/create-file/create-file.js

```diff
@@ -0,0 +1,244 @@
+      optional: true,
+      async options({ prevContext }) {
+        const { nextPageToken } = prevContext;
+        let results;
+        if (this.drive === "myDrive") {
+          results = await this.googleDrive.listFiles({
+            pageToken: nextPageToken,
+            q: "mimeType = 'application/vnd.google-apps.folder'",
+          });
+        } else {
+          results = await this.googleDrive.listFiles({
+            pageToken: nextPageToken,
+            corpora: "drive",
+            driveId: this.drive,
+            includeItemsFromAllDrives: true,
+            supportsAllDrives: true,
+            q: "mimeType = 'application/vnd.google-apps.folder'",
+          });
+        }
+        return results;
+      },
+    },
+    shortcutDetailsTargetId: {
+      type: "string",
+      label: "Shortcut Details Target ID",
+      description: "The ID of the file that this shortcut points to.",
+      optional: true,
+    },
+    starred: {
+      type: "boolean",
+      label: "Starred",
+      description: "Whether the user has starred the file.",
+      optional: true,
+    },
+    writersCanShare: {
+      type: "boolean",
+      label: "Writers Can Share",
+      description:
+        "Whether users with only writer permission can modify the file's permissions. Not populated for items in shared drives.",
+      optional: true,
+    },
+  },
+  async run() {
+    const drive = this.googleDrive.drive();
+    const body = this.fileUrl
+      ? await got.stream(this.fileUrl)
+      : fs.createReadStream(`${this.filePath}`);
+    return (
+      await drive.files.create({
+        ignoreDefaultVisibility: this.ignoreDefaultVisibility,
+        includePermissionsForView: this.includePermissionsForView,
+        keepRevisionForever: this.keeprevisionForever,
+        ocrLanguage: this.ocrLanguage,
+        useContentAsIndexableText: this.useContentAsIndexableText,
+        supportsAllDrives: this.supportsAllDrives,
+        resource: {
+          name: this.name,
+          originalFilename: this.originalFilename,
+          parents: [
+            this.parents,
+          ],
+          mimeType: this.mimeType,
+          description: this.description,
+          folderColorRgb: this.folderColorRgb,
+          shortcutDetails: {
+            targetId: this.shortcutDetailsTargetId,
+          },
+          starred: this.starred,
+          writersCanShare: this.writersCanShare,
+          contentHints: {
+            indexableText: this.contentHintsIndexableText,
+          },
+          contentRestrictions: {
+            readOnly: this.contentRestrictionsReadOnly,
+            reason: this.contentRestrictionsReason,
+          },
+          copyRequiresWriterPermission: this.copyRequiresWriterPermission,
+        },
+        media: {
+          mimeType: this.mimeType,
+          uploadType: this.uploadType,
+          body,
+        },
+        fields: "*",
+      })
+    ).data;
```

Comment:
This call should live inside a method in the app module, like we do with event sources.

Reply:
Updated to use method in the app module